core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/WorkflowSubmissionActor.scala

```diff
@@ -273,17 +276,24 @@ trait WorkflowSubmission extends FutureSupport with LazyLogging with MethodWiths
         //The workspace's billing project
         billingProject <- dataAccess.rawlsBillingProjectQuery.load(RawlsBillingProjectName(workspaceRec.namespace)).map(_.get)
 
-        //The person who submitted the submission, and their token
+        //The person who submitted the submission
         submitter <- dataAccess.rawlsUserQuery.load(RawlsUserRef(RawlsUserSubjectId(submissionRec.submitterId))).map(_.get)
-        petSAJson <- DBIO.from(samDAO.getPetServiceAccountKeyForUser(billingProject.projectName.value, submitter.userEmail))
-        petUserInfo <- DBIO.from(googleServicesDAO.getUserInfoUsingJson(petSAJson))
-        userCredentials <- DBIO.from(googleServicesDAO.getUserCredentials(submitter)).map(_.getOrElse(throw new RawlsException(s"cannot find credentials for $submitter")))
 
-        //The wdl
+        //the method configuration, in order to get the wdl
         methodConfig <- dataAccess.methodConfigurationQuery.loadMethodConfigurationById(submissionRec.methodConfigurationId).map(_.get)
+      } yield {
+        (wfRecs, workflowBatch, billingProject, submitter, methodConfig)
+      }
+    }
 
+    val workflowBatchFuture = dbThingsFuture flatMap { case (wfRecs, workflowBatch, billingProject, submitter, methodConfig) =>
```

Comment:
can `dbThingsFuture` be in the for comp too?

Reply:
Yes, I just felt that having an explicitly different looking block for the db stuff separate to external calls would make it clear that we shouldn't muddle them

Reply:
I don't think that effect has been achieved.